library/src/main/java/com/google/maps/android/clustering/algo/NonHierarchicalViewBasedAlgorithm.java

```diff
@@ -54,7 +55,23 @@ public void onCameraChange(CameraPosition cameraPosition) {
 
     @Override
     protected Collection<QuadItem<T>> getClusteringItems(PointQuadTree<QuadItem<T>> quadTree, int discreteZoom) {
-        return quadTree.search(getVisibleBounds(discreteZoom));
+        Bounds visibleBounds = getVisibleBounds(discreteZoom);
+        Collection<QuadItem<T>> items = new ArrayList<>();
+
+        // Handle wrapping around international date line
+        if (visibleBounds.minX < 0) {
+            Bounds wrappedBounds = new Bounds(visibleBounds.minX + 1, 1, visibleBounds.minY, visibleBounds.maxY);
```

Comment:
```suggestion
            Bounds wrappedBounds = new Bounds(visibleBounds.minX % 1, 1, visibleBounds.minY, visibleBounds.maxY);
```

Reply:
In practice, this will do the same thing. Overlapping the international date line can yield numbers as negative as -0.33. After that, it flips positive again. It's impossible to get a number less than -1, as this would indicate wrapping around the world multiple times.

Reply:
sgtm